### PR TITLE
chore: rename SimpleCacheClient -> CacheClient

### PR DESCRIPTION
**Pull request by pgautier404**

### CODE REVIEW

**Review by poppoerika — APPROVED** [review]
Looks good to me 🍀

File: src/Cache/CacheClient.php [hunk 1]
```diff
@@ -38,7 +37,7 @@
 use Psr\Log\LoggerAwareInterface;
```
Comment (cprice404) [inline src/Cache/CacheClient.php:37]: any idea what all the qodana stuff below is?  i feel like we should either disable that or add whatever it is doing to our CI linting?  doesn't have to be in this PR but it does make it kinda hard to review :)

**Review by cprice404 — APPROVED** [review]
would love to sort the qodana thing but not a blocker for this PR

File: src/Cache/CacheClient.php (hunk 1 above)
Comment (pgautier404) [reply src/Cache/CacheClient.php:37]: No idea, but these are clearly false positives. I downloaded the binary and it's definitely going to take some digging to see what's up with this thing. I'll go ahead and disable it for now until I have some time to take a closer look.

**Pull request merged**